libraries/AP_HAL/utility/Socket.cpp

```diff
@@ -74,16 +80,58 @@ void SocketAPM::make_sockaddr(const char *address, uint16_t port, struct sockadd
 bool SocketAPM::connect(const char *address, uint16_t port)
 {
     struct sockaddr_in sockaddr;
+    int ret;
     make_sockaddr(address, port, sockaddr);
 
-#if AP_NETWORKING_BACKEND_CHIBIOS
-    if (::lwip_connect(fd, (struct sockaddr *)&sockaddr, sizeof(sockaddr)) != 0) {
-#else
-    if (::connect(fd, (struct sockaddr *)&sockaddr, sizeof(sockaddr)) != 0) {
+    if (datagram && is_multicast_address(sockaddr)) {
+        /*
+          connect fd_in as a multicast UDP socket
+         */
+        fd_in = CALL_PREFIX(socket)(AF_INET, SOCK_DGRAM, 0);
+        if (fd_in == -1) {
```

Comment:
Does this need to be a failure here?

Reply:
it means we're out of sockets, and if we continue then we won't get any replies as the destination IP/port of the reply won't match the outgoing socket